pkg/daemon/cleanup/cleanup.go

```diff
@@ -62,16 +62,23 @@ func Deconfigure(cfg *rest.Config, ctx context.Context, configMapNamespace strin
 			cmData, err = servicescm.Parse(cm.Data)
 			return err
 		}
-		// Otherwise, fetch the ConfigMap tied to the node's desired version annotation
-		desiredVersion, present := node.Annotations[metadata.DesiredVersionAnnotation]
+		// Otherwise, fetch the ConfigMap tied to the node's version annotation.
+		// This ensures we cleanup using the same ConfigMap version that was used to configure the instance
+		version, present := node.Annotations[metadata.VersionAnnotation]
 		if !present {
-			return fmt.Errorf("node %s missing desired version annotation", node.Name)
+			// If no version annotation exists, try the desired version. This is useful if node configuration fails and
+			// the instance needs to be reconciled
+			version, present = node.Annotations[metadata.DesiredVersionAnnotation]
+			if !present {
+				return fmt.Errorf("node %s missing desired version annotation", node.Name)
+			}
 		}
 		err = directClient.Get(ctx,
-			client.ObjectKey{Namespace: configMapNamespace, Name: servicescm.NamePrefix + desiredVersion}, cm)
+			client.ObjectKey{Namespace: configMapNamespace, Name: servicescm.NamePrefix + version}, cm)
 		if err != nil {
 			return err
 		}
+		klog.Infof("removing services tied to version: %s", version)
```

Comment:
nit: this is being logged to stdout and IMO will be noise when trying to debug the stdout/err stream on WMCO failures. At the least this should be debug level

Reply:
I'm not seeing debug log capability with klog: https://github.com/kubernetes/klog/blob/v1.0.0/klog.go#L111